### ISSUE TITLE
Openresolv is no longer supported on Ubuntu 24.04 LTS

### ISSUE THREAD

**Issue by ikhwanperwira** [fscarmen/warp-sh#66]
Based on this chosen answer of support issue: https://askubuntu.com/questions/1513526/what-happened-to-openresolv-on-ubuntu-24-04-package-openresolv-has-no-instal

`openresolv` is no longer supported, instead it must be use default `systemd-resolv`. It might be better to use default config.

**Comment by fscarmen** [comment]
ubuntu24.04, I'm going to take some time to see how to handle this, do you have a solution please?

**Comment by ikhwanperwira** [comment]
> ubuntu24.04, I'm going to take some time to see how to handle this, do you have a solution please?

Nope, currently possible solution is downgrading to 22.04.

**Comment by fscarmen** [comment]
I just tested it and Ubuntu 24.04 installs and works correctly!
```
root@instance-20240602-144154:~# bash menu.sh -e

 All dependencies already exist and do not need to be installed additionally. 

 Checking VPS infomation... 
 The script specifically adds WARP network interface for VPS, detailed:[https://github.com/fscarmen/warp-sh]
 Features:
         • Support WARP+ account. Third-party scripts are use to increase WARP+ quota or upgrade kernel.
         • Not only menus, but commands with option.
         • Support system: Ubuntu 16.04、18.04、20.04、22.04,Debian 9、10、11,CentOS 7、8、9, Alpine, Arch Linux 3.
         • Support architecture: AMD,ARM and s390x
         • Automatically select four WireGuard solutions. Performance: Kernel with WireGuard integration > Install kernel module > wireguard-go
         • Suppert WARP Linux client.
         • Output WARP status, IP region and asn
 
======================================================================================================================

 Version:3.0.7
 New features:Support CentOS 9 / Alma Linux 9 / Rocky Linux 9 system.
 System infomation:
         Operating System:Ubuntu 24.04 LTS
         Kernel:6.8.0-1007-gcp
         Architecture:amd64
         Virtualization:google 
         IPv4: 34.170.50.138 United States  Google LLC 
         IPv6:     
         WARP Interface is off 
         Client is not installed. 
         WireProxy is not installed. 

======================================================================================================================

 1.  Add WARP IPv4 interface to IPv4 only VPS (bash menu.sh 4) 
 2.  Add WARP IPv6 interface to IPv4 only VPS (bash menu.sh 6) 
 3.  Add WARP dualstack interface to IPv4 only VPS (bash menu.sh d) 
 4.  Turn on WARP (warp o) 
 5.  Install CloudFlare Client and set mode to Proxy (bash menu.sh c) 
 6.  Change the WARP IP to support Netflix (warp i) 
 7.  Turn off, uninstall WARP interface, Linux Client and WireProxy (warp u) 
 8.  Getting WARP+ quota by scripts (warp p) 
 9.  Upgrade kernel, turn on BBR, change Linux system (warp b) 
 10. Sync the latest version (warp v) 
 11. Professional one-click script for WARP to unblock streaming media (Supports multi-platform, multi-mode and TG push) 
 12. Install iptable + dnsmasq + ipset. Let WARP only take over the streaming media traffic (Not available for ipv6 only) (bash menu.sh e) 
 13. Install wireproxy. Wireguard client that exposes itself as a socks5 proxy or tunnels (bash menu.sh w) 
 14. Install CloudFlare Client and set mode to WARP (bash menu.sh l) 
 0.  Exit 

 Choose: 3

 Install using:
 1. wireguard kernel (default)
 2. wireguard-go with reserved 

 Choose: 

 Working mode:
 1. Global (default)
 2. Non-global 

 Choose: 

 Is there a WARP+ or Teams account?
 1. Use free account (default)
 2. WARP+
 3. Teams 

 Choose: 

 Please choose the priority:
 1. IPv4
 2. IPv6
 3. Use initial settings (default) 

 Choose: 

 Step 1/3: Install dependencies... 

Hit:1 http://us-central1.gce.archive.ubuntu.com/ubuntu noble InRelease
Get:2 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB]
Get:3 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports InRelease [126 kB]
Get:4 http://security.ubuntu.com/ubuntu noble-security InRelease [126 kB]
Get:5 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/universe amd64 Packages [15.0 MB]
Get:6 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/universe Translation-en [5982 kB]
Get:7 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/universe amd64 Components [3871 kB]
Get:8 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/universe amd64 c-n-f Metadata [301 kB]
Get:9 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/multiverse amd64 Packages [269 kB]
Get:10 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/multiverse Translation-en [118 kB]
Get:11 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/multiverse amd64 Components [35.0 kB]
Get:12 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/multiverse amd64 c-n-f Metadata [8328 B]
Get:13 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/main amd64 Packages [96.9 kB]
Get:14 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/main Translation-en [27.9 kB]
Get:15 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/main amd64 Components [8564 B]
Get:16 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/universe amd64 Packages [44.9 kB]
Get:17 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/universe Translation-en [17.0 kB]
Get:18 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/universe amd64 Components [45.0 kB]
Get:19 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/universe amd64 c-n-f Metadata [112 B]
Get:20 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/restricted amd64 Components [212 B]
Get:21 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/multiverse amd64 Components [212 B]
Get:22 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-updates/multiverse amd64 c-n-f Metadata [116 B]
Get:23 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/main amd64 Components [208 B]
Get:24 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/main amd64 c-n-f Metadata [112 B]
Get:25 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/universe amd64 Packages [5812 B]
Get:26 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/universe Translation-en [2152 B]
Get:27 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/universe amd64 Components [17.6 kB]
Get:28 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/universe amd64 c-n-f Metadata [116 B]
Get:29 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/restricted amd64 Components [216 B]
Get:30 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/restricted amd64 c-n-f Metadata [116 B]
Get:31 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/multiverse amd64 Components [212 B]
Get:32 http://us-central1.gce.archive.ubuntu.com/ubuntu noble-backports/multiverse amd64 c-n-f Metadata [116 B]
Get:33 http://security.ubuntu.com/ubuntu noble-security/main amd64 Packages [88.0 kB]
Get:34 http://security.ubuntu.com/ubuntu noble-security/main Translation-en [24.5 kB]       
Get:35 http://security.ubuntu.com/ubuntu noble-security/main amd64 Components [6876 B]
Get:36 http://security.ubuntu.com/ubuntu noble-security/universe amd64 Packages [34.9 kB]
Get:37 http://security.ubuntu.com/ubuntu noble-security/universe Translation-en [13.4 kB]
Get:38 http://security.ubuntu.com/ubuntu noble-security/universe amd64 Components [8632 B]
Get:39 http://security.ubuntu.com/ubuntu noble-security/universe amd64 c-n-f Metadata [112 B]
Get:40 http://security.ubuntu.com/ubuntu noble-security/restricted amd64 Components [212 B]
Get:41 http://security.ubuntu.com/ubuntu noble-security/multiverse amd64 Components [208 B]
Get:42 http://security.ubuntu.com/ubuntu noble-security/multiverse amd64 c-n-f Metadata [116 B]
93% [5 Packages store 0 B]                                        
 Step 2/3: WARP is ready 

Fetched 26.4 MB in 7s (4003 kB/s)                                                                                                                 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
23 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package openresolv is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'openresolv' has no installation candidate
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  wireguard-tools
0 upgraded, 1 newly installed, 0 to remove and 23 not upgraded.
Need to get 89.1 kB of archives.
After this operation, 330 kB of additional disk space will be used.
Get:1 http://us-central1.gce.archive.ubuntu.com/ubuntu noble/main amd64 wireguard-tools amd64 1.0.20210914-1ubuntu4 [89.1 kB]
Fetched 89.1 kB in 0s (730 kB/s)     
Selecting previously unselected package wireguard-tools.
(Reading database ... 71953 files and directories currently installed.)
Preparing to unpack .../wireguard-tools_1.0.20210914-1ubuntu4_amd64.deb ...
Unpacking wireguard-tools (1.0.20210914-1ubuntu4) ...
Setting up wireguard-tools (1.0.20210914-1ubuntu4) ...
wg-quick.target is a disabled or a static unit, not starting it.
Processing triggers for man-db (2.12.0-4build2) ...
Scanning processes... [=======================================================================================                                    ]
 Step 3/3: Searching for the best MTU value and endpoint address are ready. 

Scanning processes...                                                                                                                              
Scanning linux images...                                                                                                                           

Running kernel seems to be up-to-date.

No services need to be restarted.

No containers need to be restarted.

No user sessions are running outdated binaries.

No VM guests are running outdated hypervisor (qemu) binaries on this host.
--2024-06-02 14:44:46--  https://gitlab.com/fscarmen/warp/-/raw/main/wireguard-go/wireguard-go-linux-amd64-20230223
Resolving gitlab.com (gitlab.com)... 172.65.251.78
Connecting to gitlab.com (gitlab.com)|172.65.251.78|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2691072 (2.6M) [application/octet-stream]
Saving to: ‘/usr/bin/wireguard-go’

/usr/bin/wireguard-go                100%[=====================================================================>]   2.57M  14.5MB/s    in 0.2s    

2024-06-02 14:44:46 (14.5 MB/s) - ‘/usr/bin/wireguard-go’ saved [2691072/2691072]

 Create shortcut [warp] successfully 
 Running WARP 
 Maximum 5 attempts to get WARP IP...
 Try 1 
 Got the WARP Free IP successfully, Working mode: Global 

==============================================================

 IPv4: 104.28.215.220 United States  Cloudflare, Inc. 
 IPv6: 2a09:bac5:7522:1cd2::2df:2b United States  Cloudflare Warp 
 Congratulations! WARP Free is turned on. Spend time:31 seconds.
 The script runs today: 29032. Total:28138845 
 IPv6 priority , Working mode: Global 

==============================================================

 Run again with warp [option] [lisence], such as 

 warp h (help)
 warp n (Get the WARP IP)
 warp o (Turn off WARP temporarily)
 warp u (Turn off and uninstall WARP interface and Socks5 Linux Client)
 warp b (Upgrade kernel, turn on BBR, change Linux system)
 warp a (Change account to Free, WARP+ or Teams)
 warp p (Getting WARP+ quota by scripts)
 warp v (Sync the latest version)
 warp r (Connect/Disconnect WARP Linux Client)
 warp 4/6 (Add WARP IPv4/IPv6 interface)
 warp d (Add WARP dualstack interface IPv4 + IPv6)
 warp c (Install WARP Linux Client and set to proxy mode)
 warp l (Install WARP Linux Client and set to WARP mode)
 warp i (Change the WARP IP to support Netflix)
 warp e (Install Iptables + dnsmasq + ipset solution)
 warp w (Install WireProxy solution)
 warp y (Connect/Disconnect WireProxy socks5)
 warp k (Switch between kernel and wireguard-go-reserved)
 warp g (Switch between warp global and non-global)
 warp s 4/6/d (Set stack proiority: IPv4 / IPv6 / VPS default)
```

**Comment by ikhwanperwira** [comment]
Okay it seems there is issue on GCP.
Because I fail on maximum retry:
```
 Create shortcut [warp] successfully 
 Running WARP 
 Maximum 5 attempts to get WARP IP...
 Try 1 
 ...
 Try 5
 ```
 
 There is possibly that I setup gVirtIO exclusively from Google Service that prevent me using this script.

**Comment by fscarmen** [comment]
I also tested successfully with gcp and I turned the firewall all the way on. There are no settings for gVirtIO, I don't even know where to set it up.

**Comment by ikhwanperwira** [comment]
> I also tested successfully with gcp and I turned the firewall all the way on. There are no settings for gVirtIO, I don't even know where to set it up.

Sorry I mean gVNIC

**Comment by ikhwanperwira** [comment]
Here is my log:
```
 Version:3.0.7
 New features:Support CentOS 9 / Alma Linux 9 / Rocky Linux 9 system.
 System infomation:
         Operating System:Ubuntu 24.04 LTS
         Kernel:6.8.0-1007-gcp
         Architecture:amd64
         Virtualization:google
         IPv4: 34.126.178.79 Singapore  Google LLC
         IPv6:
         WARP Interface is off
         Client is not installed.
         WireProxy is not installed.

======================================================================================================================

 1.  Add WARP IPv4 interface to IPv4 only VPS (bash menu.sh 4)
 2.  Add WARP IPv6 interface to IPv4 only VPS (bash menu.sh 6)
 3.  Add WARP dualstack interface to IPv4 only VPS (bash menu.sh d)
 4.  Turn on WARP (warp o)
 5.  Install CloudFlare Client and set mode to Proxy (bash menu.sh c)
 6.  Change the WARP IP to support Netflix (warp i)
 7.  Turn off, uninstall WARP interface, Linux Client and WireProxy (warp u)
 8.  Getting WARP+ quota by scripts (warp p)
 9.  Upgrade kernel, turn on BBR, change Linux system (warp b)
 10. Sync the latest version (warp v)
 11. Professional one-click script for WARP to unblock streaming media (Supports multi-platform, multi-mode and TG push)
 12. Install iptable + dnsmasq + ipset. Let WARP only take over the streaming media traffic (Not available for ipv6 only) (bash menu.sh e)
 13. Install wireproxy. Wireguard client that exposes itself as a socks5 proxy or tunnels (bash menu.sh w)
 14. Install CloudFlare Client and set mode to WARP (bash menu.sh l)
 0.  Exit

 Choose: 3

 Install using:
 1. wireguard kernel (default)
 2. wireguard-go with reserved

 Choose:

 Working mode:
 1. Global (default)
 2. Non-global

 Choose:

 Is there a WARP+ or Teams account?
 1. Use free account (default)
 2. WARP+
 3. Teams

 Choose:

 Please choose the priority:
 1. IPv4
 2. IPv6
 3. Use initial settings (default)

 Choose:

 Step 1/3: Install dependencies...

Warning: The unit file, source configuration file or drop-ins of apt-news.service changed on disk. Run 'systemctl daemon-reload' to reload units.
Warning: The unit file, source configuration file or drop-ins of esm-cache.service changed on disk. Run 'systemctl daemon-reload' to reload units.
Hit:1 http://asia-southeast1.gce.archive.ubuntu.com/ubuntu noble InRelease
Hit:2 http://asia-southeast1.gce.archive.ubuntu.com/ubuntu noble-updates InRelease
Hit:3 http://asia-southeast1.gce.archive.ubuntu.com/ubuntu noble-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu noble-security InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
2 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package openresolv is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'openresolv' has no installation candidate
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done

 Step 2/3: WARP is ready

Recommended packages:
  nftables | iptables
The following NEW packages will be installed:
  wireguard-tools
0 upgraded, 1 newly installed, 0 to remove and 2 not upgraded.
Need to get 89.1 kB of archives.
After this operation, 330 kB of additional disk space will be used.
Get:1 http://asia-southeast1.gce.archive.ubuntu.com/ubuntu noble/main amd64 wireguard-tools amd64 1.0.20210914-1ubuntu4 [89.1 kB]
Fetched 89.1 kB in 0s (199 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package wireguard-tools.
(Reading database ... 55798 files and directories currently installed.)                                                 Preparing to unpack .../wireguard-tools_1.0.20210914-1ubuntu4_amd64.deb ...
Unpacking wireguard-tools (1.0.20210914-1ubuntu4) ...
Setting up wireguard-tools (1.0.20210914-1ubuntu4) ...
wg-quick.target is a disabled or a static unit, not starting it.                                                        Processing triggers for man-db (2.12.0-4build2) ...
--2024-06-03 20:54:32--  https://gitlab.com/fscarmen/warp/-/raw/main/wireguard-go/wireguard-go-linux-amd64-20230223
Resolving gitlab.com (gitlab.com)... 172.65.251.78
Connecting to gitlab.com (gitlab.com)|172.65.251.78|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2691072 (2.6M) [application/octet-stream]
Saving to: ‘/usr/bin/wireguard-go’

/usr/bin/wireguard-go         100%[=================================================>]   2.57M  --.-KB/s    in 0.05s

2024-06-03 20:54:33 (52.8 MB/s) - ‘/usr/bin/wireguard-go’ saved [2691072/2691072]


 Step 3/3: Searching for the best MTU value and endpoint address are ready.

 Create shortcut [warp] successfully
 Running WARP
 Maximum 5 attempts to get WARP IP...
 Try 1
 Try 2
 Try 3
 Try 4
 Try 5
 ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
 Operating System: Ubuntu 24.04 LTS

 Kernel:6.8.0-1007-gcp

 Menu choose: 3.  Add WARP dualstack interface to IPv4 only VPS (bash menu.sh d)

 Warning: `/etc/wireguard/warp.conf' is world accessible
[#] ip link add warp type wireguard
[#] wg setconf warp /dev/fd/63
[#] ip -4 address add 172.16.0.2/32 dev warp
[#] ip -6 address add 2606:4700:110:8155:64ae:4c9f:69c1:47c1/128 dev warp
[#] ip link set mtu 1380 up dev warp
[#] resolvconf -a warp -m 0 -x
[#] wg set warp fwmark 51820
[#] ip -6 rule add not fwmark 51820 table 51820
[#] ip -6 rule add table main suppress_prefixlength 0
[#] ip -6 route add ::/0 dev warp table 51820
[#] ip6tables-restore -n
/usr/bin/wg-quick: line 32: ip6tables-restore: command not found
[#] resolvconf -d warp -f
[#] ip -6 rule delete table 51820
[#] ip -6 rule delete table main suppress_prefixlength 0
[#] ip link delete dev warp
 ↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑↑
 There have been more than 5 failures. The script is aborted. Attach the above error message. Feedback: [https://github.com/fscarmen/warp-sh/issues]
 ```

**Comment by fscarmen** [comment]
https://chatgpt.com/share/3cdc3547-1a9a-4bc6-86e5-44b032937fde

The error message you are encountering, `/usr/bin/wg-quick: line 32: ip6tables-restore: command not found`, indicates that the `ip6tables-restore` command is not available on your system. This command is part of the `iptables` package, which is used for setting up, maintaining, and inspecting the IP firewall rules in the Linux kernel.

To resolve this issue, you need to ensure that `iptables` and its IPv6 counterpart are installed on your system. You can install the required packages by running the appropriate commands for your Linux distribution.

### For Debian-based distributions (e.g., Ubuntu):

```bash
sudo apt update
sudo apt install iptables ip6tables
```

### For Red Hat-based distributions (e.g., CentOS, Fedora):

```bash
sudo yum install iptables-services
sudo systemctl enable ip6tables
sudo systemctl start ip6tables
```

After installing the necessary packages, try running your `wg-quick` command again.

### Additional Steps (if needed):

1. **Verify Installation:**
   Ensure that `ip6tables` is properly installed by running:
   ```bash
   ip6tables --version
   ```

2. **Check Path:**
   Make sure that the `ip6tables` command is in your system's PATH:
   ```bash
   which ip6tables
   ```

By following these steps, you should be able to resolve the error and successfully run `wg-quick`.

**Comment by ikhwanperwira** [comment]
I see, the problem is on my OS version since I'm using **Minimal Ubuntu**. You might add `iptables` installation on your script. First it will check if iptables already installed.


There is no such `ip6tables` package, it's already included on `iptables` on `Ubuntu 24.04 LTS`.
```
root@bastion:~# sudo apt update
sudo apt install iptables ip6tables
Hit:1 http://asia-southeast1.gce.archive.ubuntu.com/ubuntu noble InRelease
Get:2 http://asia-southeast1.gce.archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB]
Hit:3 http://asia-southeast1.gce.archive.ubuntu.com/ubuntu noble-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu noble-security InRelease
Fetched 126 kB in 1s (169 kB/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
2 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package ip6tables
```

So, just do:
```
apt install iptables
```

Thanks, it's worked.
```
 Step 3/3: Searching for the best MTU value and endpoint address are ready.

 Create shortcut [warp] successfully
 Running WARP
 Maximum 5 attempts to get WARP IP...
 Try 1
 Got the WARP Free IP successfully, Working mode: Global

==============================================================

 IPv4: 104.28.222.47 Singapore  Cloudflare, Inc.
 IPv6: 2a09:bac5:55fc:101e::19b:27 Singapore  Cloudflare Warp
 Congratulations! WARP Free is turned on. Spend time:26 seconds.
 The script runs today: 28669. Total:28185056
 IPv6 priority , Working mode: Global

==============================================================
```

The last question I want to ask is, can you give me docs what "Working Mode Global and Non-global" do?

**Comment by fscarmen** [comment]
I would only test it on an LTS standard system, the Minimal version may have all sorts of problems.
